packages/valory/skills/dynamic_nft_abci/handlers.py

```diff
@@ -279,27 +280,39 @@ def _handle_get_health(
         last_update_time = self.synchronized_data.last_update_time
 
         if last_update_time:
-            current_time = cast(
+            current_time_tm = cast(
                 SharedState, self.context.state
             ).round_sequence.abci_app.last_timestamp.timestamp()
 
+            current_time = datetime.datetime.now().timestamp()
+
             observation_interval = self.context.params.observation_interval
 
             seconds_since_last_reset = current_time - last_update_time
             seconds_until_next_update = (
                 observation_interval - seconds_since_last_reset
             )  # this can be negative if we have passed the estimated reset time without resetting
-            is_healthy = seconds_since_last_reset < 2 * observation_interval
+
+            seconds_since_last_tm_update = current_time - current_time_tm
+
+            is_healthy = all(
+                [
+                    seconds_since_last_reset < 2 * observation_interval,
+                    seconds_since_last_tm_update < 2 * observation_interval,
+                ]
+            )
```

Comment:
For recovery we use `block_stall_deadline_expired` to check whether the tm communication is healthy, ex. from `BaseBehaviour`.
```
    @property
    def tm_communication_unhealthy(self) -> bool:
        """Return if the Tendermint communication is not healthy anymore."""
        return cast(
            SharedState, self.context.state
        ).round_sequence.block_stall_deadline_expired
```
That will return true if the deadline is expired, which means we haven't received a block for 1 minute.
 https://github.com/valory-xyz/open-autonomy/blob/main/packages/valory/skills/abstract_round_abci/behaviour_utils.py#L542